kernel/Tupfile.lua

```diff
@@ -1,9 +1,9 @@
 if tup.getconfig("NO_FASM") ~= "" then return end
 tup.rule("echo lang fix " .. ((tup.getconfig("LANG") == "") and "en" or tup.getconfig("LANG")) .. " > %o", {"lang.inc"})
-tup.rule("echo __REV = 0x`git rev-parse --short HEAD` > %o", {"rev.inc"})
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o ", "bootbios.bin")
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o -dextended_primary_loader=1", "bootbios.bin.ext_loader")
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o -dpretest_build=1", "bootbios.bin.pretest")
+tup.rule("echo -n `git rev-parse HEAD` > %o", {"rev.inc"})

```

Comment:
Please use a regular shell redirect here (`git rev-parse HEAD > %o`), it will make the rule valid for windows.